src/components/sidebar/nodeView/NodeView.tsx

```diff
@@ -126,46 +131,43 @@ class NodeView extends ViewFormBase<INodeViewProps, INodeViewState> {
         this.setState({ isLoading: false });
     }
 
-    private onNodePropertiesChange = (
-        property: string,
-    ) => (
-        value: any,
-        validationResult?: IValidationResult,
-    ) => {
-        this.props.nodeStore!.updateNode(property, value);
-        if (validationResult) {
-            this.markInvalidProperties(property, validationResult!.isValid);
+    private toggleIsEditingEnabled = () => {
+        this.toggleIsEditingDisabled(
+            this.props.nodeStore!.resetChanges,
+        );
+    }
+
+    private _validateAllProperties = (nodeType: NodeType) => {
+        const node = this.props.nodeStore!.node;
+        if (nodeType === NodeType.STOP) {
+            this.validateAllProperties(stopValidatorModel, node.stop);
+        } else {
+            this.validateAllProperties({}, node);
         }
     }
 
-    private onNodeGeometryChange = (
-        property: NodeLocationType,
-    ) => (
-        value: any, validationResult?: IValidationResult,
-    ) => {
+    private onNodeGeometryChange = (property: NodeLocationType) => (value: any) => {
         this.props.nodeStore!.updateNodeGeometry(property, value);
-        if (validationResult) {
-            this.markInvalidProperties(property, validationResult!.isValid);
-        }
+        this.validateProperty({}, property, value);
     }
 
-    private onStopChange =
-        (property: string) => (value: any, validationResult?: IValidationResult) => {
-            this.props.nodeStore!.updateStop(property, value);
-            if (validationResult) {
-                this.markInvalidProperties(property, validationResult!.isValid);
-            }
+    private onNodePropertiesChange = (property: string) => (value: any) => {
+        this.props.nodeStore!.updateNode(property, value);
+        this.validateProperty({}, property, value);
+        if (property === 'type') {
+            this._validateAllProperties(value);
         }
+    }
 
-    private toggleIsEditingEnabled = () => {
-        this.toggleIsEditingDisabled(
-            this.props.nodeStore!.resetChanges,
-        );
+    private onStopPropertiesChange = (property: string) => (value: any) => {
```

Comment:
onStopPropertyChange? we are modifying one property

Reply:
We are modifiying multiple properties. This function is just used once

Reply:
(and we have onNodePropertiesChange)

Reply:
Or hmm, maybe I see your point. Changing one property at once. Then also onNodePropertiesChange should be renamed to onNodePropertyChange